### PR TITLE
`azurerm_function_app` - `storage_connection_string` can now rotate keys but still a require recreate when changing the account

### LINKED ISSUE #5435: Changing the azurerm_function_app storage_connection_string causes a recreate when it shouldn't
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.19
+ provider.azurerm v1.41.0

### Affected Resource(s)

* `azurerm_function_app`

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  location = "eastus"
  name     = "function-stg-test"
}

resource "azurerm_storage_account" "example" {
  account_replication_type = "LRS"
  account_tier             = "Standard"
  location                 = azurerm_resource_group.example.location
  name                     = "funcstgtest92783"
  resource_group_name      = azurerm_resource_group.example.name
}

resource "azurerm_app_service_plan" "example" {
  location            = azurerm_resource_group.example.location
  name                = "function-stg-test-asp-23232"
  resource_group_name = azurerm_resource_group.example.name
  kind                = "functionapp"

  sku {
    tier = "Dynamic"
    size = "Y1"
  }
}

resource "azurerm_function_app" "example" {
  app_service_plan_id       = azurerm_app_service_plan.example.id
  location                  = azurerm_app_service_plan.example.location
  name                      = "function-stg-test928732"
  resource_group_name       = azurerm_app_service_plan.example.resource_group_name
  storage_connection_string = azurerm_storage_account.example.primary_blob_connection_string
//    storage_connection_string = azurerm_storage_account.example.secondary_blob_connection_string
}
```

### Expected Behavior

I expect the `storage_connection_string` to change without disrupting the existing Function App.

### Actual Behavior

The Function App is marked and run as a recreate when only the `storage_connection_string` has changed. There isn't a comment in the plan that states the `storage_connection_string` is the cause of the recreate either.

### Steps to Reproduce

1. `terraform apply`
2. Change the connection string being used from primary to secondary.
3. `terraform apply`

### Important Factoids

Testing in normal Azure, East US

### PR DESCRIPTION
**Pull request by kraihn**
Fixes #5435

### CODE REVIEW

File: azurerm/internal/services/web/resource_arm_function_app.go [hunk 1]
```diff
@@ -78,7 +78,6 @@ func resourceArmFunctionApp() *schema.Resource {
 			"storage_connection_string": {
 				Type:      schema.TypeString,
 				Required:  true,
-				ForceNew:  true,
 				Sensitive: true,
 			},
```
Comment (tombuildsstuff) [inline azurerm/internal/services/web/resource_arm_function_app.go]: whilst this'd work it'd cause a bunch of runtime errors should someone opt to change the account being used - as such I'm wondering if we'd be better to split this field into two:

1) "storage_account_id" for the ID of the Storage Account in question - which is ForceNew
2) the "storage account access key" which is either the primary/secondary key and allows updating

WDYT?

File: azurerm/internal/services/web/resource_arm_function_app.go [hunk 2]
```diff
@@ -869,3 +878,14 @@ func flattenFunctionAppSiteCredential(input *web.UserProperties) []interface{} {
 
 	return append(results, result)
 }
+
+func getAccountName(storageConnectionString string) string {
```
Comment (tombuildsstuff) [inline azurerm/internal/services/web/resource_arm_function_app.go]: in Golang functions are shared across the entire package - as such this method would probably be better named as:

```suggestion
func getAccountNameFromStorageConnectionString(storageConnectionString string) string {
```

File: azurerm/internal/services/web/resource_arm_function_app.go (hunk 1 above)
Comment (kraihn) [reply azurerm/internal/services/web/resource_arm_function_app.go]: I love that idea! It makes more sense to me. Should I make this change with 2.0 in mind and remove the old property?

Comment (katbyte) [reply azurerm/internal/services/web/resource_arm_function_app.go]: now that we are post 2.0 what we will have to do is deprecate the old one and make them conflict.

**Comment by katbyte** [comment]
@kraihnm, going to close this in favour of #6304. thanks for opening the PR!

**Pull request closed without merging**